### PR TITLE
Add sunraster package for solar spectrogram data handling

### PR DESCRIPTION
**Pull request by DanRyanIrish**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by DanRyanIrish** [comment]
I am willing to be a recipe maintainer.

File: recipes/sunraster/meta.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  license: BSD-2-Clause
+  license_file: LICENSE.rst
+  summary: A package providing instrument-agnostic data classes for manipulating solar spectrogram data.
+  license_family: BSD
+  dev_url: https://github.com/sunpy/ndcube
+  doc_url: https://sunraster.readthedocs.io/
+
+extra:
+  recipe-maintainers:
+    - DanRyanIrish
```
Comment (Cadair) [inline recipes/sunraster/meta.yml:56]:
```suggestion
    - DanRyanIrish
    - Cadair
```

File: recipes/sunraster/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    - setuptools_scm
+  host:
+    - python
+    - pip
+  run:
+    - python >=3.7
+    - astropy >=3.1
+    - matplotlib >=2
+    - sunpy >=1.0.3
+    - ndcube >=1.3.2
```
Comment (DanRyanIrish) [inline recipes/sunraster/meta.yaml:32]: Include only packages explicitly imported in sunraster here.

File: recipes/sunraster/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - python >=3.7
+    - setuptools
+    - setuptools_scm
+  host:
+    - python
+    - pip
```
Comment (DanRyanIrish) [inline recipes/sunraster/meta.yaml:26]: Combine build and host sections into host

Comment (DanRyanIrish) [inline recipes/sunraster/meta.yaml:26]: Combine build and host sections into host

File: recipes/sunraster/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  commands:
+    - pytest --pyargs sunraster
+
+about:
+  home: http://sunpy.org
+  license: BSD-2-Clause
+  license_file: LICENSE.rst
+  summary: A package providing instrument-agnostic data classes for manipulating solar spectrogram data.
+  license_family: BSD
+  dev_url: https://github.com/sunpy/ndcube
```
Comment (synapticarbors) [inline recipes/sunraster/meta.yaml:47]: I think it should be `dev_url: https://github.com/sunpy/sunraster`

File: recipes/sunraster/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  sha256: {{ sha256 }}
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python
```
Comment (synapticarbors) [inline recipes/sunraster/meta.yaml:21]: I think the error you are having might be from having a version restriction under `run` that's missing in `host`

**Comment by DanRyanIrish** [comment]
Thanks for your comments @synapticarbors.  I've made those changes and hopefully they'll resolve those errors :)

**Pull request merged**